build.gradle

```diff
@@ -0,0 +1,190 @@
+plugins {
+	id 'java'
+	id 'war'
+	id 'jacoco'
+	id 'org.ajoberstar.grgit' version '4.1.1'
+}
+
+// TODO NEXT build script - how include built jar
+// TODO NEXT update dockerfile
+// TODO NEXT remove ant & makefile
+// TODO NEXT move to standard gradle locations
+// TODO NEXT run tests from Eclipse w/o specifying classpath manually & remove sourceSets & classpath
+// TODO NEXT use jitpack jars vs pulling from repo
+
+// TODO TEST DEPS TypeGeneratorTest has a hard coded list of jars it uses for tests
+//           Need to figure out a better way
+// TODO TEST add a test that starts the SDK in a docker container and checks some simple cmds
+// TODO DEPS make JitPack jars for catalog and NMS and use them
+
+repositories {
+	mavenCentral()
+	maven {  // for syslog4j
+		name = "Clojars"
+		url = "https://repo.clojars.org/"
+	}
+}
+
+task buildGitCommitFile {
+	doLast {
+		def commitId = grgit.head().id
+		// is there a variable for builddir/classes/java/main?
+		file("$buildDir/classes/java/main/us/kbase/mobu/gitcommit").text = commitId
+	}
+}
+
+compileJava {
+	options.release = 11
+	finalizedBy buildGitCommitFile
+}
+
+task pythonTest(type: Exec) {
+	description = 'Runs Python tests using pytest'
+	environment 'PYTHONPATH', './src/java/us/kbase/templates'
```

Comment:
This seems a bit dodgy -- presumably because the various client modules are in that folder, though. There doesn't seem much point in moving them elsewhere just to make the path cleaner.

TL;DR: just me thinking to myself.

Reply:
yeah, the makefile does the same thing. You have to tell python where your code under test lives, so I don't really see a way around this